pkg/mlsstore/store.go

```diff
@@ -125,6 +121,59 @@ func (s *Store) ConsumeKeyPackages(ctx context.Context, installationIds []string
 	return keyPackages, nil
 }
 
+func (s *Store) GetIdentityUpdates(ctx context.Context, walletAddresses []string, startTimeNs int64) (map[string]IdentityUpdateList, error) {
+	updated := make([]*Installation, 0)
+	// Find all installations that were changed since the startTimeNs
+	err := s.db.NewSelect().
+		Model(&updated).
+		Where("wallet_address IN (?)", bun.In(walletAddresses)).
+		WhereGroup(" AND ", func(q *bun.SelectQuery) *bun.SelectQuery {
+			return q.Where("created_at > ?", startTimeNs).WhereOr("revoked_at > ?", startTimeNs)
+		}).
+		Order("created_at ASC").
+		Scan(ctx)
```

Comment:
If I understood correctly, the server just returns a list of installations with their status (registered/revoked)? And we're not attaching the original signed payloads?

So the issue with this approach is we're relying 100% on server trust, not just for ordering but for determining who to encrypt messages for. The server can add and remove arbitrary clients for any wallet address, which circumvents the E2EE.

Reply:
The server can definitely hide arbitrary installations. Adding a fake installation will not get the server very far. The purpose of this endpoint is to get a concise list of `installation_id`s which the client may then request key packages for. If the key package doesn't exist, or doesn't validate correctly, it would be ignored and could be flagged to the user.

Revocation isn't fully fleshed out. We are going to do want to include proof for that once we figure out what the revocation payloads actually look like. The protos are setup to allow us to add more fields to the revocation updates once we know what to put in there.

Reply:
I see, so this can work. My next question here is about overall design. In order to verify the server's result here, the client MUST then fetch the corresponding `KeyPackages`. Essentially, the current design assumes that we will never want to fetch a `Credential` without fetching a `KeyPackage`.

 Here are the types of query patterns that I anticipate we will need:
 
 1. What are my active installations, so that I can revoke them if needed?
 2. What are the other users' active installations in this conversation, so that I can verify the membership is correct?
 3. I want to start a conversation with another user - give me all of their key packages.
 
I believe we only want KeyPackages for (3), and for (1) and (2) we don't necessarily want to consume any one-time keypackages. We could issue the query for LastResortKeyPackages specifically, but that seems inelegant. Naively, I would think (1) and (2) could be cleaner by fetching Credentials and Revocations directly, rather than an unsigned list of installations provided by the server?

Reply:
We synced offline. I am going to update things to include both the credential and public key on the installation, so that we can include it in `GetIdentityUpdates` for client verification.